libraries/resource_providers/api200/connection_template_provider.rb

```diff
@@ -15,19 +15,29 @@ module OneviewCookbook
   module API200
     # ConnectionTemplate API200 provider
     class ConnectionTemplateProvider < ResourceProvider
-      def load_resource_from_ethernet
-        return unless @context.associated_ethernet_network
+      def load_template_from_resource(resource_type, resource_name)
+        return unless resource_name
         @item.data.delete('name')
-        @item['uri'] = load_resource(:EthernetNetwork, @context.associated_ethernet_network, 'connectionTemplateUri')
+        @item['uri'] = load_resource(resource_type, resource_name, :connectionTemplateUri)
+      end
+
+      def load_connection_templates
+        resources_set = [@context.associated_ethernet_network, @context.associated_fcoe_network, @context.associated_fc_network,
+                         @context.associated_network_set].reduce(0) { |sum, value| (value ? 1 : 0) + sum } == 1
+        raise 'A single associated resource field must be specified for this action.' unless resources_set
```

Comment:
It should be `> 1` since we can just update the connection template directly without getting it from an associated resource.
Liked the solution though, didn't know about `compact`. @tmiotto and I had to bump heads a little to come up with the map&reduce for this.